### DISCUSSION TITLE
Shard non divisible batch size

### DISCUSSION THREAD

**Discussion by long21wt** [google/flax#2914] [Q&A] [answered]
Hi, 

I'd like to sample my batch according to Poisson sampling, therefore the batch size might result in an indivisible number of local devices to be shard.

- Normal Sampling 
   - devices = 2, all batch size = 10 → OK
- Poisson Sampling
  - devices = 2, a single batch size = 10 → OK
  - devices = 2, a single batch size = 5 → Failed
 
Is there a workaround for my case?
Thanks

**Comment by cgarciae** [comment, marked as answer]
Hey @long21wt, when using batch sizes that are not divisible by the number of devices the solution is usually to use form some of padding + masking. Checkout our [Processing the entire Datase](https://flax.readthedocs.io/en/latest/guides/full_eval.html) guide for some pointers.

**Comment by long21wt** [comment]
Thanks, seems like padding is the easiest way.
